client/MonitoringServiceApiCaller/AutoRegistrationInMonitoring.cs

```diff
@@ -30,33 +33,64 @@ public static class AutoRegistrationInMonitoring
         {
             if (configuration == null)
                 throw new ArgumentNullException(nameof(configuration));
+
+            string disableAutoRegistrationStr = configuration[_disableAutoRegistrationEnvVarName];
+            if (bool.TryParse(disableAutoRegistrationStr, out bool disableAutoRegistration) && disableAutoRegistration)
+                return;
+
             if (string.IsNullOrWhiteSpace(monitoringServiceUrl))
                 throw new ArgumentException("Argument is empty", nameof(monitoringServiceUrl));
+
             if (log == null)
-                log = new LogToConsole();
+                throw new ArgumentNullException(nameof(log));
+
+            string podTag = configuration[_podNameEnvVarName] ?? "";
+
             try
             {
                 string myMonitoringUrl = configuration[_myMonitoringUrlEnvVarName];
                 if (string.IsNullOrWhiteSpace(myMonitoringUrl))
                 {
                     myMonitoringUrl = _missingEnvVarUrl;
-                    log.WriteInfo("Auto-registration in monitoring", "", $"{_myMonitoringUrlEnvVarName} environment variable is not found. Using {myMonitoringUrl} for monitoring registration");
+                    log.WriteMonitor(
+                        "Auto-registration in monitoring",
+                        podTag,
+                        $"{_myMonitoringUrlEnvVarName} environment variable is not found. Using {myMonitoringUrl} for monitoring registration");
                 }
                 string myMonitoringName = configuration[_myMonitoringNameEnvVarName];
                 if (string.IsNullOrWhiteSpace(myMonitoringName))
                     myMonitoringName = PlatformServices.Default.Application.ApplicationName;
                 var monitoringService = new MonitoringServiceFacade(monitoringServiceUrl);
+
+                try
+                {
+                    var monitoringRegistration = await monitoringService.GetService(myMonitoringName);
+                    if (monitoringRegistration.Url == myMonitoringUrl)
+                        return;
+
+                    log.WriteMonitor("Auto-registration in monitoring", podTag, $"There is a registration for {myMonitoringName} in monitoring service!");
+
+                    myMonitoringUrl = _missingEnvVarUrl;
+                    if (string.IsNullOrEmpty(podTag))
+                        podTag = Guid.NewGuid().ToString();
```

Comment:
Either declare another variable (i.e. podName), or implement this inititalization logic at line 47, where podTag is defined.